bindings/profiler.cc

```diff
@@ -85,73 +82,142 @@ NAN_METHOD(StopSamplingHeapProfiler) {
 
 NAN_METHOD(GetAllocationProfile) {
   std::unique_ptr<v8::AllocationProfile> profile(
-    info.GetIsolate()->GetHeapProfiler()->GetAllocationProfile());
+      info.GetIsolate()->GetHeapProfiler()->GetAllocationProfile());
   AllocationProfile::Node* root = profile->GetRootNode();
   info.GetReturnValue().Set(TranslateAllocationProfile(root));
 }
 
-
 // Time profiler
 
 #if NODE_MODULE_VERSION > NODE_8_0_MODULE_VERSION
-// This profiler exists for the lifetime of the program. Not calling 
+// This profiler exists for the lifetime of the program. Not calling
 // CpuProfiler::Dispose() is intentional.
 CpuProfiler* cpuProfiler = CpuProfiler::New(v8::Isolate::GetCurrent());
 #else
 CpuProfiler* cpuProfiler = v8::Isolate::GetCurrent()->GetCpuProfiler();
 #endif
 
-
-Local<Value> TranslateTimeProfileNode(const CpuProfileNode* node) {
+Local<Value> TranslateTimeProfileNode(const CpuProfileNode* node,
+                                      bool hasLines) {
   Local<Object> js_node = Nan::New<Object>();
   js_node->Set(Nan::New<String>("name").ToLocalChecked(),
-    node->GetFunctionName());
+               node->GetFunctionName());
   js_node->Set(Nan::New<String>("scriptName").ToLocalChecked(),
-    node->GetScriptResourceName());
+               node->GetScriptResourceName());
   js_node->Set(Nan::New<String>("scriptId").ToLocalChecked(),
-    Nan::New<Integer>(node->GetScriptId()));
+               Nan::New<Integer>(node->GetScriptId()));
   js_node->Set(Nan::New<String>("lineNumber").ToLocalChecked(),
-    Nan::New<Integer>(node->GetLineNumber()));
+               Nan::New<Integer>(node->GetLineNumber()));
   js_node->Set(Nan::New<String>("columnNumber").ToLocalChecked(),
-    Nan::New<Integer>(node->GetColumnNumber()));
+               Nan::New<Integer>(node->GetColumnNumber()));
   js_node->Set(Nan::New<String>("hitCount").ToLocalChecked(),
-    Nan::New<Integer>(node->GetHitCount()));
+               Nan::New<Integer>(node->GetHitCount()));
   int32_t count = node->GetChildrenCount();
-  Local<Array> children = Nan::New<Array>(count);
+
+  unsigned int index = 0;
+  Local<Array> children;
+
+  // Add nodes corresponding to lines within the node's function.
+  unsigned int hitLineCount = node->GetHitLineCount();
+  std::vector<CpuProfileNode::LineTick> entries(hitLineCount);
+  if (hasLines && node->GetLineTicks(&entries[0], hitLineCount)) {
+    children = Nan::New<Array>(count + entries.size());
+    js_node->Set(Nan::New<String>("hitCount").ToLocalChecked(),
```

Comment:
A comment here would be good to explain what is going on and why the number 0 is special.

Reply:
Sorry -- getting confused with github.

Which line does this comment refer to?

Reply:
`hasDetailedLines && node->GetLineTicks(&entries[0], hitLineCount`